cli/engine/ApplyEngine.py

```diff
@@ -140,7 +140,21 @@ def assert_no_postgres_nodes_number_change(self):
                 prev_postgres_count = len(existing_inventory.list_hosts(pattern='postgresql'))
                 next_postgres_count = int(components['postgresql']['count'])
 
-                if prev_postgres_count != next_postgres_count:
+                # To avoid situation where the postgres is configured on some other host different then postgres
+                if next_postgres_count == 0:
+                    postgres_role_mapping_exists = False
+
+                    for object in self.configuration_docs:
```

Comment:
Check comment on previous line, there should be no need to loop over the configuration docs for the 'configuration/feature-mapping' if its taken from default or input docs straight.